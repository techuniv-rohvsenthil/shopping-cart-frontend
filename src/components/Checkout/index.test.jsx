import React from 'react';
import { render, waitForDomChange, fireEvent } from '@testing-library/react';
import CheckOut from './index';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import axios from 'axios';

const cart = [
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
  ]
  
  jest.mock('axios');

describe('the CheckOut component', () => {
  it('should render correctly', () => {
    const history = createMemoryHistory();
    const { asFragment } = render(<Router history={history}><CheckOut  cart={cart}/></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  xit('should make api call when checkout button is clicked', async () => {
    axios.mockImplementation(() => {});
    const history = createMemoryHistory();
    const { getByTestId } = render(<Router history={history}><CheckOut  cart={cart}/></Router>);
    await waitForDomChange();
    fireEvent.click(getByTestId('checkout'));
    expect(axios).toHaveBeenCalled();
    
  });

});