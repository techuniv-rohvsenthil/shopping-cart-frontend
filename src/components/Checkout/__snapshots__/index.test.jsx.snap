// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CheckOut component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        Total
      </div>
      200
    </div>
    <div />
    <button
      data-testid="checkout"
      type="button"
    >
      Checkout
    </button>
  </div>
</DocumentFragment>
`;
